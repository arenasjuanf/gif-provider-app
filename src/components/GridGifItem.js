import PropTypes from 'prop-types';

const GridGifItem = ( {title, url} ) => {

    return(
        <div className="card animate__bounceIn">
            <img src={ url } alt={ title } />
            <p>
                { title }
            </p>
           
        </div>
    )
}

GridGifItem.propTypes = {
    url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
}

export default GridGifItem;