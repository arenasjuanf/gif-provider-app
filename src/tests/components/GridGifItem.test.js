import { shallow } from 'enzyme'
import GridGifItem from '../../components/gridGifItem';


describe("<GridGifItem /> tests", () => {

    const [ title, url ] = ['a title', 'https://localhost/some.php'];
    const wrapper = shallow(<GridGifItem title={ title } url={ url } />);

    test("it should to show successfully the component", () => {
        expect( wrapper ).toMatchSnapshot();
    })

    test("it should had a title", () => {
        const p = wrapper.find("p");
        expect( p.text().trim() ).toBe(title);
    })

    test("img in props equal to 'url'", () => {
        const img = wrapper.find("img");
        expect( img.prop('src') ).toBe(url);
    })

    test("has classname 'animate__bounceIn'", () => {
        const div = wrapper.find("div");

        expect( div.hasClass('animate__bounceIn') ).toBe(true);
    })
})