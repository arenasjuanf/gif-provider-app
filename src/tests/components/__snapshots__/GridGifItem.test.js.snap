// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridGifItem /> tests it should to show successfully the component 1`] = `
<div
  className="card animate__bounceIn"
>
  <img
    alt="a title"
    src="https://localhost/some.php"
  />
  <p>
    a title
  </p>
</div>
`;
